src/main/java/roboy/dialog/states/ordinaryStates/IntroductionState.java

```diff
@@ -0,0 +1,156 @@
+package roboy.dialog.states.ordinaryStates;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import roboy.context.Context;
+import roboy.dialog.states.definitions.State;
+import roboy.dialog.states.definitions.StateParameters;
+import roboy.linguistics.Linguistics;
+import roboy.linguistics.Triple;
+import roboy.linguistics.sentenceanalysis.Interpretation;
+import roboy.memory.Neo4jRelationships;
+import roboy.memory.nodes.Interlocutor;
+import roboy.memory.nodes.Interlocutor.RelationshipAvailability;
+import static roboy.memory.nodes.Interlocutor.RelationshipAvailability.*;
+import roboy.memory.nodes.MemoryNodeModel;
+import roboy.dialog.Segue;
+import roboy.util.RandomList;
+
+import java.util.List;
+import java.util.Set;
+
+import static roboy.memory.Neo4jRelationships.*;
+
+
+/**
+ * This state will:
+ * - ask the interlocutor for his name
+ * - query memory if the person is already known
+ * - create and update the interlocutor in the context
+ * - take one of two transitions: knownPerson or newPerson
+ *
+ * IntroductionState interface:
+ * 1) Fallback is not required.
+ * 2) Outgoing transitions that have to be defined:
+ *    - knownPerson:    following state if the person is already known
+ *    - newPerson:      following state if the person is NOT known
+ * 3) No parameters are used.
+ */
+public class IntroductionState extends State {
+    private final String UPDATE_KNOWN_PERSON = "knownPerson";
+    private final String LEARN_ABOUT_PERSON = "newPerson";
+    private final Logger logger = LogManager.getLogger();
+
+    private Neo4jRelationships[] predicates = { FROM, HAS_HOBBY, WORK_FOR, STUDY_AT };
+    private State nextState;
+
+    public IntroductionState(String stateIdentifier, StateParameters params) {
+        super(stateIdentifier, params);
+    }
+
+    @Override
+    public Output act() {
+        // TODO: add more different introduction phrases
+        return Output.say("I'm Roboy. What's your name?");
```

Comment:
would be really nice to have more than one phrase here, even if you just choose one from a hardcoded RandomList

Reply:
Refactored